src/IECoreGL/Selector.cpp

```diff
@@ -396,14 +424,28 @@ class Selector::Implementation : public IECore::RefCounted
 				throw IECore::Exception( "ID shader does not have an ieCoreGLNameOut output" );
 			}
 
+			GLint depthDataLocation = 0;
+			if( m_useCameraDepth )
+			{
+				depthDataLocation = glGetFragDataLocation( shader->program(), "ieCoreGLCameraDepth" );
+				if( depthDataLocation < 0 )
+				{
+					throw IECore::Exception( "ID shader does not have ieCoreGLCameraDepth output" );
+				}
+			}
+
 			m_nameUniformLocation = nameParameter->location;
 
 			m_currentIDShader = shader;
 			glUseProgram( m_currentIDShader->program() );
 
 			std::vector<GLenum> buffers;
-			buffers.resize( fragDataLocation + 1, GL_NONE );
-			buffers[buffers.size()-1] = GL_COLOR_ATTACHMENT0;
+			buffers.resize( std::max( fragDataLocation, depthDataLocation ) + 1, GL_NONE );
+			buffers[buffers.size() - ( depthDataLocation > 0 ? 2 : 1 )] = GL_COLOR_ATTACHMENT0;
```

Comment:
Is the indexing right here? It looks like the original code could guarantee that the last element was the one we wanted to set because we were resizing to `fragDataLocation + 1`. But in a user-provided custom shader we have no control over the locations, and they could each now be anywhere between 0 and `max( fragDataLocation, depthDataLocation )`. I _think_ this is what we want, unless I'm missing something :

```
buffers[fragDataLocation] = GL_COLOR_ATTACHMENT0;
buffers[depthDataLocation] = GL_COLOR_ATTACHMENT1;
```

Reply:
Right, the indexing is not correct here. Fixed in https://github.com/ImageEngine/cortex/pull/1404/commits/91a36463965e3010731a5443f6e18c2f009b3586. I think we still need the conditional so we don't risk stomping over another buffer if we're not using `depthDataLocation`.